presto-orc/src/main/java/com/facebook/presto/orc/reader/ByteStreamReader.java

```diff
@@ -255,7 +252,7 @@ public void scan()
                 }
                 nextActive = inputPositions[activeIdx];
                 if (outputChannelSet && numResults > resultSizeBudget) {
-                    truncationRow = inputQualifyingSet.truncateAndReturnTruncationRow(activeIdx);
+                    batchTooLarge(activeIdx);
```

Comment:
it would be clear to have this method return an exception and have the caller throw: `throw batchTooLarge();`